plan/sort_test.go

```diff
@@ -320,3 +320,310 @@ func TestSortComplex(t *testing.T) {
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d2": {
+								Name:    "d2",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+			expected: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "1s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d2": {
+								Name:    "d2",
+								Metrics: []string{"m2"},
+							},
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "2 levels, freq not multiples, distinct domain, distinct metrics",
+			input: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "2s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d2": {
+								Name:    "d2",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+			expected: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "2s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d2": {
+								Name:    "d2",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "2 levels, freq not multiples, shared domain, distinct metrics",
+			input: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "2s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+			expected: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "2s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
`t.Run` isn't necessary here. Just exec the contents of this func directly in the test.